### DISCUSSION TITLE
How to install Playlet as a sideloaded/dev app

### DISCUSSION THREAD

**Discussion by iBicha** [iBicha/playlet#371] [General]
<p align="center">
  <img src="https://raw.githubusercontent.com/iBicha/playlet/main/docs/banner.png" style="display:block;text-align:center" alt="Playlet Logo" width="800"/>
</p>

## Note: [Playlet is available on the Roku channel store](https://channelstore.roku.com/details/4a41d0921265a5e31429a7679442153f:b5bcb5b630c28b01e93bf59856317b43/playlet)
Or using access code: https://my.roku.com/account/add?channel=PLAYLET

## You can also find Playlet by searching "Playlet" in the channel store on your Roku device.
_If you installed Playlet from the store, you do not need to sideload the app._

If you cannot install Playlet from the store for some reason (for example, you cannot find it in the store in your country), you can sideload it as a dev app.

Please note that Roku only supports one sideloaded app at a time. If you have another sideloaded app installed, you need to make a choice between it and Playlet.

Another important thing to note is that sideloaded apps share the locally stored data and cache. If you install another sideloaded app after Playlet, it might be able to read your profile, your settings, or even your **Invidious access token** and **YouTube access token**.

With that being said:

### 1. Enable developer settings
If you haven't already, see [instructions to enable developer mode in your Roku device](https://github.com/iBicha/playlet/tree/main/docs#roku-developer-mode)

### 2. Download Playlet
Go to https://github.com/iBicha/playlet/releases and find the release that says "latest". Then download `playlet.zip`
Here's a shortcut to that file: https://github.com/iBicha/playlet/releases/latest/download/playlet.zip

<img width="955" alt="release" src="https://github.com/iBicha/playlet/assets/17722782/42c452ec-6cbd-4935-814a-a50cb23f76b6">

### 3. Install Playlet
Find out your Roku's ip address from the network settings. There's plenty of guides online, but here's one https://www.howtogeek.com/806547/how-to-find-roku-ip-address/

Open the address in your browser. Enter `rokudev` for the username, and enter the password that you chose when you enabled developer settings.

You should then see the **Development Application Installer**
<img width="985" alt="rokudev" src="https://github.com/iBicha/playlet/assets/17722782/05888705-d304-4cb4-87cd-b54247c6f2d7">
Click `Upload` and choose `playlet.zip` that you downloaded from the previous step. Then click `Install with zip`.

Once installed, you should see a message indicating a successful install, and Playlet should launch automatically.

<img width="830" alt="success" src="https://github.com/iBicha/playlet/assets/17722782/712d0138-ab6c-4315-b30c-60fbcd153bcb">

### 4. Updating Playlet
Playlet downloads most of its functionality from `playlet-lib` on startup using Github. So there's no need to do frequent updates. But it's a good idea to update it every once in a while, since dev channels do not get updated automatically.

**Comment by garionphx** [comment]
How often do you update the version in the Roku store? I think mine is on 0.25 still... I was gonna open a ticket about missing preview images, but realized I'm on a really old version.

**Reply by iBicha** [reply]
> How often do you update the version in the Roku store? I think mine is on 0.25 still... I was gonna open a ticket about missing preview images, but realized I'm on a really old version.

I do not update the app on the store often, it's been many months since the last update.
0.25 is fine, it doesn't matter because the app is a thin loader that pulls the latest release from GitHub. The app's job is to download a library and run it. That functionality is simple and doesn't need frequent updates.

"Playlet lib" is where the functionality is, and it gets frequent updates.

If you have any problem, just open a GitHub issue and we will get to the bottom of it

**Reply by afrank84** [reply]
Thanks for all your hard work, just found out about this repo, it is amazing. @iBicha

**Comment by nolanweakley921-debug** [comment]
PLAYLET
